src/dropdown.ts

```diff
@@ -215,48 +218,37 @@ export class Dropdown extends Component<DropdownOptions> implements Openable {
     e.preventDefault();
     this._moveDropdown((<HTMLElement>e.target).closest('li'));
     if (this.isOpen) {
-        this.close();
+      this.close();
     } else {
-        this.open();
+      this.open();
     }
-  }
+  };
 
   _handleMouseEnter = (e) => {
     this._moveDropdown((<HTMLElement>e.target).closest('li'));
     this.open();
-  }
+  };
 
   _handleMouseLeave = (e: MouseEvent) => {
     const toEl = e.relatedTarget as HTMLElement;
     const leaveToDropdownContent = !!toEl.closest('.dropdown-content');
     let leaveToActiveDropdownTrigger = false;
     const closestTrigger = toEl.closest('.dropdown-trigger');
-    if (
-      closestTrigger &&
-      !!(<any>closestTrigger).M_Dropdown &&
-      (<any>closestTrigger).M_Dropdown.isOpen
-    ) {
+    if (closestTrigger && !!closestTrigger['M_Dropdown'] && closestTrigger['M_Dropdown'].isOpen) {
```

Comment:
I think we might need to revert this, since it introduces backwards compatibility issues with applications using the previous object orientated approach, propose to just set it to warnings until we can declare the correct type definition